src/QiParsers.h

```diff
@@ -52,6 +52,7 @@ inline bool parseNumber(char decimalMark, char groupingMark, Iterator& first,
   double sum = 0, denom = 1;
   NumberState state = STATE_INIT;
   bool seenNumber = false;
+  bool negative = *first == '-';
```

Comment:
Ok it is so (https://github.com/hadley/readr/pull/420/commits/163f780ce61c1d9379e64dc43f5b77a73fc5b145)